### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by punkymaniac**

### CODE REVIEW

File: include/git2/revwalk.h [hunk 1]
```diff
@@ -277,6 +278,7 @@ GIT_EXTERN(git_repository *) git_revwalk_repository(git_revwalk *walk);
  *
  * @param commit_id oid of Commit
  * @param payload User-specified pointer to data to be passed as data payload
+ * @return non-zero to hide the commmit and it parrent.
```
Comment (ethomson) [inline include/git2/revwalk.h:281]:
```suggestion
 * @return non-zero to hide the commmit and it parent.
```

File: include/git2/revwalk.h [hunk 2]
```diff
@@ -249,6 +249,7 @@ GIT_EXTERN(int) git_revwalk_push_range(git_revwalk *walk, const char *range);
  *
  * No parents other than the first for each commit will be enqueued.
  *
+ * @param walk The revisionwlaker.
```
Comment (ethomson) [inline include/git2/revwalk.h:252]:
```suggestion
 * @param walk The revision walker.
```

File: include/git2/repository.h [hunk 3]
```diff
@@ -928,6 +931,7 @@ GIT_EXTERN(int) git_repository_is_shallow(git_repository *repo);
  * @param name where to store the pointer to the name
  * @param email where to store the pointer to the email
  * @param repo the repository
+ * @return 0
```
Comment (ethomson) [inline include/git2/repository.h:934]:
```suggestion
 * @return 0 or an error code
```

File: include/git2/rebase.h [hunk 4]
```diff
@@ -322,6 +326,10 @@ GIT_EXTERN(int) git_rebase_next(
  * This is only applicable for in-memory rebases; for rebases within
  * a working directory, the changes were applied to the repository's
  * index.
+ *
+ * @param index The result index of the last operation.
+ * @param rebase The in-progress rebase.
+ * @return 0
```
Comment (ethomson) [inline include/git2/rebase.h:332]:
```suggestion
 * @return 0 or an error code
```

File: include/git2/odb.h [hunk 5]
```diff
@@ -345,6 +346,11 @@ GIT_EXTERN(int) git_odb_stream_finalize_write(git_oid *out, git_odb_stream *stre
  * Read from an odb stream
  *
  * Most backends don't implement streaming reads
+ *
+ * @param stream the stream
+ * @param buffer the buffer where to store the readed data.
```
Comment (ethomson) [inline include/git2/odb.h:351]:
```suggestion
 * @param buffer a user-allocated buffer to store the data in.
```

File: include/git2/object.h [hunk 6]
```diff
@@ -221,6 +221,7 @@ GIT_EXTERN(int) git_object_peel(
  *
  * @param dest Pointer to store the copy of the object
  * @param source Original object to copy
+ * @return 0
```
Comment (ethomson) [inline include/git2/object.h:224]:
```suggestion
 * @return 0 or an error code
```

File: include/git2/indexer.h [hunk 7]
```diff
@@ -132,6 +136,8 @@ GIT_EXTERN(int) git_indexer_commit(git_indexer *idx, git_indexer_progress *stats
  * names. This is only correct after the index has been finalized.
  *
  * @param idx the indexer instance
+ * @return the packfile's hash. In casewhere the index has not been finalized,
+ *	it will be zeroed out.
```
Comment (ethomson) [inline include/git2/indexer.h:140]: We state that this should only be called when it's been finalized, let's not make promises about what happens if you don't obey that.

```suggestion
 * @return the packfile's hash
```

File: include/git2/config.h [hunk 8]
```diff
@@ -735,6 +750,7 @@ GIT_EXTERN(int) git_config_parse_path(git_buf *out, const char *value);
  * @param regexp regular expression to match against config names (can be NULL)
  * @param callback the function to call on each variable
  * @param payload the data to pass to the callback
+ * @return non-zero to terminate the iteration.
```
Comment (ethomson) [inline include/git2/config.h:753]: This comment is really about how the callback behaves, not the function itself.

```suggestion
 * @return 0 or an error code.
```

File: include/git2/config.h [hunk 9]
```diff
@@ -437,6 +442,7 @@ GIT_EXTERN(int) git_config_get_multivar_foreach(const git_config *cfg, const cha
  * @param name the variable's name
  * @param regexp regular expression to filter which variables we're
  * interested in. Use NULL to indicate all
+ * @return non-zero to terminate the iteration.
```
Comment (ethomson) [inline include/git2/config.h:445]:
```suggestion
 * @return 0 or an error code.
```

File: include/git2/config.h [hunk 10]
```diff
@@ -422,6 +426,7 @@ GIT_EXTERN(int) git_config_get_string_buf(git_buf *out, const git_config *cfg, c
  * interested in. Use NULL to indicate all
  * @param callback the function to be called on each value of the variable
  * @param payload opaque pointer to pass to the callback
+ * @return non-zero to terminate the iteration.
```
Comment (ethomson) [inline include/git2/config.h:429]: This explains how `git_config_foreach_cb` should stop callbacks, it doesn't describe the return value of the function.

```suggestion
 * @return 0 or an error code.
```

File: include/git2/branch.h [hunk 11]
```diff
@@ -34,6 +34,8 @@ GIT_BEGIN_DECL
  *
  * @param out Pointer where to store the underlying reference.
  *
+ * @param repo the repository where to create the branch.
```
Comment (ethomson) [inline include/git2/branch.h:37]: A little awkward, would rephrase as:

```suggestion
 * @param repo the repository to create the branch in.
```

File: include/git2/commit.h [hunk 12]
```diff
@@ -479,6 +479,7 @@ GIT_EXTERN(int) git_commit_create_buffer(
  * to the commit and write it into the given repository.
  *
  * @param out the resulting commit id
+ * @param repo the repository where to create the commit.
```
Comment (ethomson) [inline include/git2/commit.h:482]: A little awkward.

```suggestion
 * @param repo the repository to create the commit in.
```

File: include/git2/attr.h [hunk 13]
```diff
@@ -349,6 +353,11 @@ GIT_EXTERN(int) git_attr_cache_flush(
  * macro, you would call:
  *
  *     git_attr_add_macro(repo, "binary", "-diff -crlf");
+ *
+ * @param repo The repository where to add the macro.
```
Comment (ethomson) [inline include/git2/attr.h:357]: A little awkward.

```suggestion
 * @param repo The repository to add the macro in.
```

**Review by ethomson — COMMENTED** [review]
Thanks for tackling this!  Just a few minor suggestions.  Let me know if you have questions or disagree with any of them, or would like to refine anything.  If not, I'll merge this 🔜.

Thanks again!

**Comment by punkymaniac** [comment]
Sorry about the delay.
Thanks for merging and adding some minor improvement.

But the function `git_object_dup` and `git_repository_indent` will never return an error code.
So the change made on the return value is incorrect.

**Comment by ethomson** [comment]
> But the function git_object_dup and git_repository_indent will never return an error code.
So the change made on these return value is incorrect.

That's an implementation detail.  They _will not_ today, but they _may_ in the future and users should expect that they could.
